tests/python/test_python_frontend.py

```diff
@@ -1639,11 +1637,37 @@ def fusion_set_func(fd: FusionDefinition):
             self.assertEqual(eager_out, nvf_out[0])
 
             nvf_stride = nvf_out[0].stride()
-            sorted_stride = list(nvf_stride)
-            rank = len(nvf_stride)
-            for idx, axis in enumerate(perm):
-                sorted_stride[rank - 1 - axis] = nvf_stride[idx]
-            self.assertTrue(sorted(sorted_stride, reverse=True) == sorted_stride)
+            verify_stride_order(nvf_stride, perm)
+
+    def test_output_stride_order_with_reduction(self):
+        inputs = [torch.randn(2, 3, 4, 5, device="cuda", dtype=torch.float)]
+
+        for stride_order in itertools.permutations(range(3), 3):
+
+            def fusion_add_output(fd: FusionDefinition) -> None:
+                T0 = fd.from_pytorch(inputs[0])
+                T1 = fd.ops.sum(T0, dims=[2])
+                fd.add_output(T1, stride_order=stride_order)
+
+            with FusionDefinition() as fd:
+                fusion_add_output(fd)
+
+            out = fd.execute(inputs)[0]
+            verify_stride_order(out.stride(), stride_order)
+            self.assertEqual(out.is_contiguous(), is_trivial_stride_order(stride_order))
```

Comment:
This is a strange check.

i.e. pytorch check for `is_contiguous` needs a context of the given memory format. I think verifying stride above should be enough.

Reply:
Hmm, I am missing something here.
If the set `stride_order` is non-trivial, the output tensor generated is non-contiguous. Hence, why I set the `contiguity` flag above. What is the `contiguity` flag representing then?

Reply:
I'll discuss this offline with you, which might be easier for a higher bandwidth conversation.

But for the record here, the contiguity flag in nvfuser merely means the indexing for the given dimension can be collapsed to its inner dimension (skipping broadcast/reduction dimensions).
This is how we compute contiguity e.g. https://github.com/NVIDIA/Fuser/blob/748d91fd93eefc796651407068ebdccebe4d3944/csrc/python_frontend/python_bindings.cpp#L414-L460

Reply:
BTW, the application of this is to simplify indexing.